### DISCUSSION TITLE
Is it possible to render emojis as well?

### DISCUSSION THREAD

**Discussion by the-jan-peter** [simonhaenisch/md-to-pdf#107] [Q&A] [answered]
I really love the tool but I ran in an inconvenience where Emojis where still rendered as text and not as graphics.

`:star:` --> :star:

Is it possible to enable this somehow?
I searched but I was unsuccessful.

**Comment by simonhaenisch** [comment, marked as answer]
Hey, unicode emoji are supported in general, i. e. if you put in ⭐️ instead of `:star:`, it should work. That double-colon syntax is not converted automatically though by `marked` (which is the underlying tool to convert the markdown to html)... 

It's possible to extend the `marked` renderer to do that; giving an example here but not claiming that it's robust in this way (feel free to figure out sth better), and I haven't tried running this, so not sure it actually works in this way.

```js
// config.js

const marked = require('marked');
const { emojify } = require('node-emoji');

const renderer = new marked.Renderer();

renderer.text = emojify;

module.exports = {
  // other options...
  marked_options: { renderer },
};
```

```
md-to-pdf --config-file config.js
```

You can refer to https://marked.js.org/using_pro to get more info about extending `marked`. There's also another option called `walkTokens` so this might also work:

```js
module.exports = {
  marked_options: {
    walkTokens: (token) => {
      token.text = emojify(token.text);
      // might also need
      // token.tokens = this.Lexer.lexInline(token.text);
    }
  }
}
```

---

TBH it would be nice to have a `marked` plugin for this and allow to set a list of plugins to use in the config. It's also possible to do that because of this new [custom extensions](https://marked.js.org/using_pro#extensions) feature, but I currently don't have the time to implement it.

**Reply by the-jan-peter** [reply]
I see.
Thank you for the quick and extensive help!

There are so much more ideas in our minds than hours in our hands ...
